### ISSUE TITLE
Fix for building with native OpenAL on OS X

### ISSUE THREAD

**Issue by moubctez** [FrodeSolheim/fs-uae#92]
FS-UAE should use native OpenAL on OS X. First of all, do not depend on a specific SDK, but use OpenAL.framework. This assures correct building on different OS X and Xcode versions.

```
--- configure.orig	2015-11-07 13:22:13.000000000 +0000
+++ configure
@@ -7122,7 +7122,7 @@ fi
 $as_echo "$as_me: Using OpenAL.framework" >&6;}
 		# OPENAL_CFLAGS="-I/System/Library/Frameworks/OpenAL.framework/Headers"
 		# OPENAL_CFLAGS="-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/System/Library/Frameworks/OpenAL.framework/Headers"
-		OPENAL_CFLAGS="-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenAL.framework/Headers"
+		OPENAL_CFLAGS="-framework OpenAL"
 		OPENAL_LIBS="-framework OpenAL"
 
 	 ;; #(
@@ -7148,7 +7148,7 @@ $as_echo "no" >&6; }
 $as_echo "$as_me: Using OpenAL.framework" >&6;}
 		# OPENAL_CFLAGS="-I/System/Library/Frameworks/OpenAL.framework/Headers"
 		# OPENAL_CFLAGS="-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/System/Library/Frameworks/OpenAL.framework/Headers"
-		OPENAL_CFLAGS="-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenAL.framework/Headers"
+		OPENAL_CFLAGS="-framework OpenAL"
 		OPENAL_LIBS="-framework OpenAL"
 
 	 ;; #(
```

Second, include OpenAL/al.h and OpenAL/alc.h headers from OpenAL framework.

```
--- libfsemu/src/emu/audio_openal.c.orig	2015-11-07 13:27:07.000000000 +0000
+++ libfsemu/src/emu/audio_openal.c
@@ -14,11 +14,15 @@
 
 #ifdef HAVE_AL_AL_H
 #include <AL/al.h>
+#elif defined(__APPLE__)
+#include <OpenAL/al.h>
 #else
 #include <al.h>
 #endif
 #ifdef HAVE_AL_ALC_H
 #include <AL/alc.h>
+#elif defined(__APPLE__)
+#include <OpenAL/alc.h>
 #else
 #include <alc.h>
 #endif
```

**Comment by rofl0r** [comment]
rather than hardcoding platform-specific assumptions, a configure check should test and set HAVE_OPENAL_ALC_H

**Comment by FrodeSolheim** [comment]
Thank you for you suggestions, changes implemented in 683e00a2591c62ff28f53d4b1da713233c23eb3d.
